### PR TITLE
fix: use normalized version constraints

### PR DESCRIPTION
**Pull request by dsherret**
* Uses https://github.com/denoland/deno_semver/pull/32

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -850,18 +847,18 @@ mod tests {
...
+      content: "{ \"version\": \"2000\" }",
+      overwrite: false,
+    })
+    .unwrap_err();
+    match err.source {
+      LockfileErrorReason::UnsupportedVersion { version } => {
+        assert_eq!(version, "2000")
+      }
+      _ => unreachable!(),
+    }
```
Comment (bartlomieju) [inline src/lib.rs:861]: Nice, see you in 4000 years 🖐️

File: src/printer.rs [hunk 2]
```diff
@@ -29,34 +29,47 @@ struct SerializedNpmPkg<'a> {
   dependencies: Vec<Cow<'a, str>>,
 }
 
+// WARNING: It's important to implement Ord/PartialOrd on the final
+// normalized string so that sorting works according to the final
+// output and so that's why this is used rather than JsrDepPackageReq
+// directly.
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Serialize)]
```
Comment (bartlomieju) [inline src/printer.rs:36]: Great, there's a warning, is there a test that asserts that as well?

**Review by bartlomieju — APPROVED** [review]
LGTM

File: src/printer.rs (hunk 2 above)
Comment (dsherret) [reply src/printer.rs:36]: I'll try to come up with one.

Comment (dsherret) [reply src/printer.rs:36]: Can't think of one atm 😅

**Pull request merged**